molecule/dynamic/playbook.yml

```diff
@@ -0,0 +1,20 @@
+---
+- hosts: all
+  pre_tasks:
+    # The version string below is the highest of all those in roles' metadata:
+    # "min_ansible_version". It needs to be kept manually up-to-date.
+    - name: Verify Ansible meets min required version
+      assert:
+        that: "ansible_version.full is version_compare('2.8', '>=')"
+        msg: >
+          "You must update Ansible to at least 2.8 to use this version of Pulp 3 Installer."
+
+    - name: Dynamically include prereq roles
+      include_role:
+        public: yes
```

Comment:
Oh, I didn't know about this option. It is newish (Ansilbe 2.7).
This 1 option may solve the entire problem that led us to want to test include_role to begin with.
Did you try leaving it out? According to 1the Katello team (and 1 or 2 other users), they had to workaround an issue by setting a variable globally ("vars:" in playbook.yml.) I forget if that issue is still present.

Reply:
without it, it didn't work, because one role was depending on the vars of other roles. So it was breaking due to "missing vars".

Reply:
Let's figure out if our users are OK with "public" or not for the long-term. I reached out to @jlsherrill . In the meantime, let's figure out that variable and try to address it as a separate PR that this is rebased on. (Or as a separate commit on this PR.) I think one technique we've used in the past is to do "loose coupling", putting the defaults in both roles that need the variable (even though the variable primarily pertains to 1 role.)

Reply:
done! Added a commit for it here!

Reply:
It's awesome that you tracked down all these missing variables. I didn't realize there were so many.

Reply:
So with all the variables added to the defaults, do we still need "public"?

Reply:
no, we don't need public, but I wasn't sure if I had to remove it, wdyt? Remove it or not?

Reply:
I decided to remove it, I think it is better for CI to see if it breaks when missing some variable